Sources/Commands/SwiftRunCommand.swift

```diff
@@ -331,36 +335,30 @@ public struct SwiftRunCommand: AsyncSwiftCommand {
         #if !os(Windows)
         // Dispatch will disable almost all asynchronous signals on its worker threads, and this is called from `async`
         // context. To correctly `exec` a freshly built binary, we will need to:
-        // 1. reset the signal masks
+        // 1. Reset the signal masks
         for i in 1..<NSIG {
             signal(i, SIG_DFL)
         }
         var sig_set_all = sigset_t()
         sigfillset(&sig_set_all)
         sigprocmask(SIG_UNBLOCK, &sig_set_all, nil)
 
-        #if os(FreeBSD) || os(OpenBSD)
-        #if os(FreeBSD)
-        pthread_suspend_all_np()
-        #endif
-        closefrom(3)
-        #else
-        #if os(Android)
-        let number_fds = Int32(sysconf(_SC_OPEN_MAX))
-        #else
-        let number_fds = getdtablesize()
-        #endif /* os(Android) */
-        
-        // 2. close all file descriptors.
-        for i in 3..<number_fds {
-            close(i)
-        }
-        #endif /* os(FreeBSD) || os(OpenBSD) */
+        // 2. Close all file descriptors above stdio.
+        closeFdsAbove(3)
         #endif
 
         try TSCBasic.exec(path: path, args: args)
     }
 
+    #if !os(Windows)
+    private func closeFdsAbove(_ from: UInt32) {
+        #if os(FreeBSD)
+        pthread_suspend_all_np()
```

Comment:
This function is only ever called after `fork()` and pre-`execve()`, right? There should be precisely one active thread so this seems like a noop.

Reply:
I was always somewhat confused on this. This is existing code though so didn't want to remove in case I was missing something..

Reply:
wow, that is weird. Tbh I would still err on the side of removing the code. If it does anything, we have a huge issue.

Reply:
https://github.com/swiftlang/swift-package-manager/pull/8327/files#r1981142987

Reply:
This might make sense if the above is true ^

Reply:
After your latest findings: Ignore this, maybe move it out of the `closeFdsAbove` function higher up (alongside a comment) because it seems important now.

Plus, tiny nit: Swift style guidelines would have `closeFDsAbove` because "fd" is an abbreviation so it's `FD` unless it's the start of a word (e.g. `fdNumber` but `highestFD`)